src/js/flows/ingestFiles.ts

```diff
@@ -148,11 +149,30 @@ const executeLoader = (
 
     // only supporting one loader at this time
     const l = loaders[0]
+    const abortCtl = new AbortController()
+
+    let didAbort = false
```

Comment:
If you wanted to avoid this variable, you could check `abortCtl.signal.aborted` 

https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal/aborted